tests/testthat/test-whoami.R

```diff
@@ -0,0 +1,25 @@
+context("whoami")
+
+test_that("whoami works in presence of PAT", {
+  skip_on_travis()
+  skip_on_appveyor()
+  res <- gh_whoami()
+  res$token <- NULL
+  expect_equivalent(
+    res,
+    list(name = "Jennifer (Jenny) Bryan",
+         login = "jennybc",
+         html_url = "https://github.com/jennybc",
+         scopes = "admin:org, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user"))
+})
+
+test_that("whoami works in absence of PAT", {
+  expect_message(res <- gh_whoami(.token = ""),
```

Comment:
Well I was using it in lookup https://github.com/jimhester/lookup/blob/e841d72819e39242e0987aa6f23b240c9d47d60c/R/rcpp.R#L3.

But I think if you are expecting a response to have a specific error class you should catch just that specific class and let any other error be signaled normally.